src/errors.rs

```diff
@@ -0,0 +1,90 @@
+use failure::{Backtrace, Context, Fail};
+use std::fmt::{self, Debug, Display};
+
+/// Kinds of errors which might happen during the generation of an archive
+#[derive(Debug, Fail)]
+pub enum CompressionErrorKind {
```

Comment:
While this seems to have a lot of boilerplate, I'm afraid it's the correct choice for this application.

Reply:
So should I fix something on that aspect ?

Reply:
Nah, let's leave this part as is.